packages/core/nuxt-theme-module/theme/components/Auth/RegisterForm.vue

```diff
@@ -0,0 +1,200 @@
+          <SfInput
+            data-cy="login-input_password"
+            v-model="form.password"
+            :valid="!errors[0]"
+            :errorMessage="errors[0]"
+            name="password"
+            label="Password"
+            type="password"
+            class="form__element"
+          />
+        </ValidationProvider>
+        <ValidationProvider :rules="{ required: { allowFalse: false } }" v-slot="{ errors }">
+          <SfCheckbox
+            v-model="createAccount"
+            :valid="!errors[0]"
+            :errorMessage="errors[0]"
+            name="create-account"
+            label="I want to create an account"
+            class="form__element"
+          />
+        </ValidationProvider>
+        <SfButton
+          data-cy="login-btn_submit"
+          type="submit"
+          class="sf-button--full-width form__button"
+          :disabled="loading"
+        >
+          <SfLoader :class="{ loader: loading }" :loading="loading">
+            <div>{{ $t('Create an account') }}</div>
+          </SfLoader>
+        </SfButton>
+      </form>
+    </ValidationObserver>
+    <div class="action">
+      {{ $t('or') }}
+      <SfButton data-cy="login-btn_login-into-account" class="sf-button--text" @click="switchAuthModal('login')">
+        {{ $t('login in to your account') }}
+      </SfButton>
+    </div>
+  </div>
+</template>
+
+<script>
+import { extend, ValidationObserver, ValidationProvider } from 'vee-validate';
+import { SfAlert, SfButton, SfCheckbox, SfInput, SfLoader } from '@storefront-ui/vue';
+import { email, required } from 'vee-validate/dist/rules';
+import { ref } from '@vue/composition-api';
+import { useUser } from '<%= options.generate.replace.composables %>';
+import { useUiState, useUiNotification } from '~/composables';
+import { authErrors } from '~/helpers/errors';
+
+extend('email', {
+  ...email,
+  message: 'Invalid email'
+});
+
+extend('required', {
+  ...required,
+  message: 'This field is required'
+});
+
+export default {
+  name: 'RegisterForm',
+  components: {
+    ValidationProvider,
+    ValidationObserver,
+    SfAlert,
+    SfButton,
+    SfCheckbox,
+    SfInput,
+    SfLoader
+  },
+  setup(_, context) {
+    const { register, loading, error } = useUser();
+    const { toggleAuthModal, switchAuthModal } = useUiState();
+    const { send } = useUiNotification();
+    const serverError = ref({});
+    const form = ref({});
+    const createAccount = ref(false);
+    const { $i18n } = context.root;
+
+    const handleError = ({ email }) => {
+      const currErr = error.value.register;
+      if (!currErr) {
+        send({
+          type: 'success',
+          message: $i18n.t('Successfully created a new account')
+        });
+        toggleAuthModal();
+        return;
+      }
+
+      const knownErrors = authErrors(context, email);
+      serverError.value = knownErrors.find(knownError => knownError.originalMessage === currErr.message);
```

Comment:
how about replacing both with just `getFriendlyError(i18n, message)`

Reply:
We can even go one step further and implement this as a UI heleper so you don't have to pass context in `setup`